src/dsll_progs/setTheory.dsl

```diff
@@ -0,0 +1,26 @@
+tconstructor Set: type 				
+tconstructor Map([a,b] : [Set(),Set()]) : type
+tconstructor Map2(['a,'b] : [type,type]) : type 	
```

Comment:
Any particular reason why we need `Map2`, for maps between objects of arbitrary types??

Reply:
Conceptually we might like to define map to be generic and not limit it to work only over sets . 
Because of the way the languages are built now, in a case you create a type which get a generic argument, you can not provide to it objects that has been defined before. For example, the following program will not work:
`Set() A
Set() B
Map2(A,B)`
But it will work with `Map(A,B)` which is defined specifically over sets. I don’t think that’s ideal, I discussed that with Josh before I left but we didn’t decide to change the design in that stage